src/js/components/WaitForSettings/WaitForSettings.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import { isActiveZoneOnCloudflare } from '../../selectors/activeZone';
+import Loading from 'cf-component-loading';
+import Text from 'cf-component-text';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+class WaitForSettings extends Component {
+
+  render() {
+    let activeZone = this.props.activeZone;
+    let analytics = (typeof this.props.analytics != 'undefined' ? this.props.analyics : true);
+    let settings = (typeof this.props.settings != 'undefined' ? this.props.settings : true);
```

Comment:
```
typeof variableThatDoesntExist
"undefined"
```
Notice it returns the string `"undefined"` not the type `undefined`.

Your comment below about switching is essentially how this is implemented.  Non-empty objects will evaluate to `true` when we evaluate:
```
 let isEverythingLoaded = settings && pluginSettings && analytics;
```

for this:
```
let analytics = (typeof this.props.analytics != 'undefined' ? this.props.analyics : true);
```
If its not `undefined` it exists, and we will use the object which will evaluate to `true` or it will be the value `false` returned from the selector if it can't find the object.
If it is `undefined` that means it wasn't passed in as a prop and we want to set it to `true` since it doesn't need to be checked.